pike/core.py

```diff
@@ -530,7 +536,34 @@ def __call__(self, cls):
         cls._register.append((self.table,self.keyattrs))
         return cls
 
-class Enum(long):
+
+class EnumMeta(type):
+    def __new__(mcs, cname, bases, idict):
+        nametoval = {}
+        valtoname = {}
+        misc = {}
+
+        for (name,val) in idict.items():
+            if name[0].isupper():
+                nametoval[name] = val
+                valtoname[val] = name
+            else:
+                misc[name] = val
+
+        cls = type.__new__(mcs, cname, bases, misc)
+        cls._nametoval = nametoval
+        cls._valtoname = valtoname
+
+        return cls
+
+    def __getattribute__(cls, name):
+        nametoval = type.__getattribute__(cls, '_nametoval')
+        if name in nametoval:
+            return cls(nametoval[name])
+        else:
+            return type.__getattribute__(cls, name)
+
+class Enum(with_metaclass(EnumMeta, int)):
```

Comment:
I'll assume that creating a new `Enum` class, as opposed to using `enum.Enum`, was a good idea.

Reply:
it's not necessarily a good idea, but `enum.Enum` wasn't around when this was written.

ideally i'd like to refactor this to use the standard `Enum` when possible